yeonnj/level1/크기가작은부분문자열.js

```diff
@@ -0,0 +1,10 @@
+function solution(t, p) {
+  const temp = [];
+
+  for (let i = 0; i <= t.length - p.length; i++) {
+    let str = t.slice(i, i + p.length);
+    temp.push(str);
+  }
+
+  return temp.filter((el) => +el <= +p).length;
+}
```

Comment:
```suggestion
function solution(t, p) {
  let result = 0;

  for (let i = 0; i <= t.length - p.length; i++) {
    const slicedNumber = Number(t.slice(i, i + p.length));
    
    if (slicedNumber <= p) {
        result++;
    }
  }

  return result;
}
```

temp라는 배열에 push하고 return문에서 filter를 진행하는게 아닌 첫 번째 for문에서 크기 비교를 진행하면 좀 더 효율적으로 진행 할 수 있을 것 같습니다 😄

Reply:
추가로 최초 코드에서 `let str = t.slice(i, i + p.length);` str 변수는 값이 변화하지 않기 때문에 `const`로 정의하는게 좋다고 판단됩니다 :) 
값이 변경되지 않은 변수를 const로 정의함으로써 예측가능한 개발을 하는데 도움이 될 수 있습니다!!

Reply:
넵 코드가 좀 더 간결해지고, 가독성에 좋은 코드인것같습니다! 참고하겠습니다 감사합니다 !